homeassistant/components/awair/sensor.py

```diff
@@ -1,248 +1,247 @@
+                if sensor in SENSOR_TYPES:
+                    sensors.append(AwairSensor(sensor, result.device, coordinator))
+
+            # The "DUST" sensor for Awair is a combo pm2.5/pm10 sensor only
+            # present on first-gen devices in lieu of separate pm2.5/pm10 sensors.
+            # We handle that by creating fake pm2.5/pm10 sensors that will always
+            # report identical values, and we let users decide how they want to use
+            # that data - because we can't really tell what kind of particles the
+            # "DUST" sensor actually detected. However, it's still useful data.
+            if API_DUST in device_sensors:
+                for alias_kind in DUST_ALIASES:
+                    sensors.append(AwairSensor(alias_kind, result.device, coordinator))
+
+    async_add_entities(sensors)
 
 
 class AwairSensor(Entity):
-    """Implementation of an Awair device."""
-
-    def __init__(self, data, device, sensor_type, throttle):
-        """Initialize the sensor."""
-        self._uuid = device[CONF_UUID]
-        self._device_class = SENSOR_TYPES[sensor_type]["device_class"]
-        self._name = f"Awair {self._device_class}"
-        unit = SENSOR_TYPES[sensor_type]["unit_of_measurement"]
-        self._unit_of_measurement = unit
-        self._data = data
-        self._type = sensor_type
-        self._throttle = throttle
+    """Defines an Awair sensor entity."""
+
+    def __init__(
+        self, kind: str, device: AwairDevice, coordinator: AwairDataUpdateCoordinator,
+    ) -> None:
+        """Set up an individual AwairSensor."""
+        LOGGER.debug("Setting up %s: %s", kind, device.uuid)
+        self._kind = kind
+        self._device = device
+        self._coordinator = coordinator
+
+        entity_id_name = SENSOR_TYPES[self.__kind][ATTR_DEVICE_CLASS]
+        if self.__device.name:
+            entity_id_name = f"{self.__device.name} {entity_id_name}"
+
+        self.entity_id = async_generate_entity_id(
+            ENTITY_ID_FORMAT, entity_id_name, hass=self.__coordinator.hass
+        )
 
     @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
+    def should_poll(self) -> bool:
+        """Return the polling requirement of the entity."""
+        return False
 
     @property
-    def device_class(self):
-        """Return the device class."""
-        return self._device_class
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        name = SENSOR_TYPES[self._kind][ATTR_LABEL]
+        if self._device.name:
+            name = f"{self._device.name} {name}"
+
+        return name
 
     @property
-    def icon(self):
-        """Icon to use in the frontend."""
-        return SENSOR_TYPES[self._type]["icon"]
+    def unique_id(self) -> str:
+        """Return the uuid as the unique_id."""
+        return f"{self._device.uuid}-{SENSOR_TYPES[self._kind][ATTR_DEVICE_CLASS]}"
```

Comment:
Does the unique_id stay the same?

Reply:
It should, unless there is some horrible bug on the Awair side of things that causes the devices UUID to change. I can't imagine why that would ever change.

Reply:
Good!